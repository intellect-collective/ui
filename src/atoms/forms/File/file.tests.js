import React from 'react';
import File from '.';
import {
    basicChangeableTests
} from '../../../../test/utils';

const validProps = {
    name: 'test',
    onClick: () => {}
};
const expectedProps = {
    name: 'test'
};

describe('File', () => {
    basicChangeableTests(File, 'input', validProps, expectedProps);
    // rerenderSuppressionTests(File, validProps);
});
