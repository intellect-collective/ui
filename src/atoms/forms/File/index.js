import React from 'react';
import changeable from '../_decorators/changeable';
import formContext from '../_decorators/formContext';
import { compose } from '../_decorators/_utils';

const File = (props) => (
    <input { ...props } type="file" />
);
File.displayName = 'File';

export default compose(
    formContext(),
    changeable()
)(File);
