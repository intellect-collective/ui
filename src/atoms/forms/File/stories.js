import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import File from './index';

storiesOf('Atoms / Forms / File', module)
    .add('default', () => (<File name="test-file-1" onChange={ action('onChange') } />));
