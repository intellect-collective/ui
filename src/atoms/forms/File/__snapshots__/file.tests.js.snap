// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Basic changeable tests Should accept valid props 1`] = `
<FormContext
  name="test"
  onClick={[Function]}
>
  <FormContextInner
    ctx={null}
    name="test"
    onClick={[Function]}
  >
    <Changeable
      name="test"
      onClick={[Function]}
    >
      <File
        name="test"
        onChange={[Function]}
        onClick={[Function]}
      >
        <input
          name="test"
          onChange={[Function]}
          onClick={[Function]}
          type="file"
        />
      </File>
    </Changeable>
  </FormContextInner>
</FormContext>
`;
